yast2_rdp.pm

```diff
@@ -0,0 +1,194 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: this test checks that YaST2's RDP module is behaving
+#          correctly and verifying that RDP service has been
+#          successfully set.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+
+my $repos = {'12-SP1' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP1/SLE-12-SP1/',
+    '12-SP1-UP' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP1/SLE-12-SP1_Update/',
+    '12-SP2'    => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP2/SLE_12_SP2/',
+    '12-SP2-UP' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP2/SUSE_SLE-12-SP2_Update/',
+    '12-SP3'    => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP3/SLE_12_SP3/',
+    '12-SP3-UP' => '',
+    '12-SP4'    => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP4/SLE_12_SP4/',
+    '12-SP4-UP' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP4/SUSE_SLE-12-SP4_Update/'
+};
+
+sub becomeroot {
+    type_string "su -\n";
+    wait_still_screen;
+    type_string "$testapi::password";
+    wait_screen_change { send_key "ret" };
+
+    script_run('test $(id -u) -eq 0 && echo "imroot" > /dev/' . $testapi::serialdev, 0);
+    wait_serial("imroot", 5) || die "Root prompt not there";
+}
+
+sub setup_system {
+    x11_start_program('xterm', valid => 0);
+
+    becomeroot;
+    # make sure that yast2-rdp has been installed.
+    if (is_sle('>12-SP2') and is_sle('<15')) {
+        my $version = get_var('VERSION');
+        zypper_ar("$repos->{$version}", 'Devel-YaST-SLE');
+        if ("$version" ne '12-SP3') {
+            zypper_ar("$repos->{qq($version-UP)}", 'Devel-YaST-SLE-UP');
+        }
+    } elsif (is_sle('<12-SP3')) {
+        die "This version of product is NOT supported.";
+    }
+    zypper_call("in yast2-rdp", exitcode => [0, 102, 103]);
+    type_string "exit\n";
+
+    # disable screen saver.
```

Comment:
why ?

Reply:
you know, when runing yast2-rdp.pm, this packages maybe not be installed, this makes sure yast2-rdp package is installed, of course, if it was installed, yast2-rdp.pm will continue.